### PR TITLE
TriggerEventT improvements

### PR DESCRIPTION
**Pull request by ElvishJerricco**
- Made `EventTriggerRef` concrete in `IORef`
- Added a `PrimMonad` instance
- Added a `MonadAdjust` instance

### CODE REVIEW

File: src/Reflex/TriggerEvent/Base.hs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE TypeFamilies #-}
+
```
Comment (ryantrinkle) [inline src/Reflex/TriggerEvent/Base.hs]: Reasonable, but should be done globally, not just in this one source file.  Also, I think a module comment would serve better.

File: src/Reflex/Class.hs [hunk 2]
```diff
@@ -999,7 +999,7 @@ class (Reflex t, Monad m) => MonadAdjust t m | m -> t where
   traverseDMapWithKeyWithAdjust :: GCompare k => (forall a. k a -> v a -> m (v' a)) -> DMap k v -> Event t (PatchDMap k v) -> m (DMap k v', Event t (PatchDMap k v'))
   traverseDMapWithKeyWithAdjustWithMove :: GCompare k => (forall a. k a -> v a -> m (v' a)) -> DMap k v -> Event t (PatchDMapWithMove k v) -> m (DMap k v', Event t (PatchDMapWithMove k v'))
 
-instance (Reflex t, MonadAdjust t m) => MonadAdjust t (ReaderT r m) where
+instance MonadAdjust t m => MonadAdjust t (ReaderT r m) where
```
Comment (ryantrinkle) [inline src/Reflex/Class.hs]: Looks good; i wonder why this wasn't a warning; GHC bug?

File: src/Reflex/TriggerEvent/Base.hs [hunk 3]
```diff
@@ -29,21 +33,28 @@ import Reflex.TriggerEvent.Class
 data TriggerInvocation a = TriggerInvocation a (IO ())
 
 -- | A reference to an 'EventTrigger' suitable for firing with 'TriggerEventT'.
-newtype EventTriggerRef t m a = EventTriggerRef { unEventTriggerRef :: Ref m (Maybe (EventTrigger t a)) }
+newtype EventTriggerRef t a = EventTriggerRef { unEventTriggerRef :: IORef (Maybe (EventTrigger t a)) }
```
Comment (ryantrinkle) [inline src/Reflex/TriggerEvent/Base.hs]: What's the benefit of monomorphizing this?

Comment (ElvishJerricco) [inline src/Reflex/TriggerEvent/Base.hs]: `m` means a lot of different things in a lot of different contexts. In `ImmediateDomBuilderT`, `m` would mean the base monad. But elsewhere, you want to be able to use `DomBuilder t m => ... -> EventTriggerRef t m -> ...` or something. If we want it to be polymorphic, it should just be polymorphic over the `Ref` type, not `m` itself.

File: src/Reflex/TriggerEvent/Base.hs (hunk 1 above)
Comment (ElvishJerricco) [inline src/Reflex/TriggerEvent/Base.hs]: Oh right. Force of habit. I'll add a commit removing this.

File: src/Reflex/TriggerEvent/Base.hs (hunk 3 above)
Comment (ryantrinkle) [inline src/Reflex/TriggerEvent/Base.hs]: That's a good point.  I'm mostly thinking that we might want to support STRef, TVar, or something else like that at some point.

File: src/Reflex/TriggerEvent/Base.hs (hunk 1 above)
Comment (ryantrinkle) [inline src/Reflex/TriggerEvent/Base.hs]: NP

**Comment by ElvishJerricco** [comment]
Are we expecting the travis to fail? The error is pretty strange.

**Comment by ryantrinkle** [comment]
Travis seems pretty unreliable here.  I don't know what's up, and haven't had a chance to debug it.

File: src/Reflex/TriggerEvent/Base.hs (hunk 3 above)
Comment (ElvishJerricco) [inline src/Reflex/TriggerEvent/Base.hs]: I don't think it'll be hard to change when that time comes. `IORef` is already hard coded in a number of places.

**Pull request merged**